src/libs/LocalePhoneNumber.js

```diff
@@ -1,53 +1,89 @@
+    callback: (val) => {
+        // When signed out, val is undefined
+        if (!val) {
+            return;
+        }
+
+        currentUserEmail = val.email;
+    },
+});
+
+let currentUserPersonalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS,
+    callback: (val) => {
+        currentUserPersonalDetails = lodashGet(val, currentUserEmail, {});
+    },
+});
+
+let countryCodeByIP;
+Onyx.connect({
+    key: ONYXKEYS.COUNTRY_CODE,
+    callback: val => countryCodeByIP = val || 1,
+});
 
 /**
- * Returns a locally converted phone number without the country code
+ * Returns a locally converted phone number for numbers from the same region
+ * and an internationally converted phone number with the country code for numbers from other regions
  *
- * @param {String} locale eg 'en', 'es-ES'
  * @param {String} number
  * @returns {String}
  */
-function toLocalPhone(locale, number) {
-    const numString = lodashTrim(number);
-    const withoutPlusNum = lodashIncludes(numString, '+') ? Str.cutBefore(numString, '+') : numString;
-    const country = lodashGet(translations, [locale, 'phoneCountryCode']);
-
-    if (country) {
-        if (lodashStartsWith(withoutPlusNum, country)) {
-            return Str.cutBefore(withoutPlusNum, country);
-        }
-        return numString;
+function formatPhoneNumber(number) {
+    const parsedPhoneNumber = parsePhoneNumber(number);
+
+    // return the string untouched if it's not a phone number
+    if (!parsedPhoneNumber.valid && !parsedPhoneNumber.possible) {
+        return number;
+    }
+
+    let signedInUserCountryCode;
+
+    /**
+     * if there is a phone number saved in the user's personal details we format the other numbers depending on
+     * the phone number's country code, otherwise we use country code based on the user's IP
+     */
+    if (currentUserPersonalDetails.phoneNumber) {
+        signedInUserCountryCode = parsePhoneNumber(currentUserPersonalDetails.phoneNumber).countryCode;
+    } else {
+        signedInUserCountryCode = countryCodeByIP;
     }
-    return number;
+
+    const regionCode = parsedPhoneNumber.countryCode;
+
+    if (regionCode === signedInUserCountryCode) {
+        return parsedPhoneNumber.number.national;
+    }
+
+    return parsedPhoneNumber.number.international;
 }
 
-/**
- * Returns an internationally converted phone number with the country code
- *
- * @param {String} locale eg 'en', 'es-ES'
- * @param {String} number
- * @returns {String}
- */
-function fromLocalPhone(locale, number) {
-    const numString = lodashTrim(number);
-    const withoutPlusNum = lodashIncludes(numString, '+') ? Str.cutBefore(numString, '+') : numString;
-    const country = lodashGet(translations, [locale, 'phoneCountryCode']);
-
-    if (country) {
-        if (lodashStartsWith(withoutPlusNum, country)) {
-            return `+${withoutPlusNum}`;
-        }
-        return `+${country}${withoutPlusNum}`;
+function formatPhoneNumberInText(text) {
```

Comment:
If we do what I said in my comment above, where we format phone numbers in IOU messages on the server, then I believe we won't need this function.

Reply:
Yeah, I totally agree with you on [this one](https://github.com/Expensify/App/pull/16804#discussion_r1163133056)! Especially if it's already used in the backend it would be nice if the numbers in the messages were sent already formatted. Should I change it now and remove the function? Or maybe should we wait until the changes are applied on the backend?

Reply:
Let's go ahead and remove it now in this PR. No need to block this PR on it. And I'll handle the back-end update.